src/coreclr/vm/interpframeallocator.cpp

```diff
@@ -0,0 +1,134 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#ifdef FEATURE_INTERPRETER
+
+#include "interpexec.h"
+#include "interpframeallocator.h"
+
+FrameDataAllocator::FrameDataFragment::FrameDataFragment(size_t size)
+{
+    if (size < INTERP_STACK_FRAGMENT_SIZE)
```

Comment:
Can you please help me understand why we need to enforce a minimum size here?

Reply:
This is simple optimization to amortize allocation cost by grouping many small allocations into a single heap block; otherwise each small allocation would call malloc. Let me know if you find it unnecessary.